cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -149,6 +182,31 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, useInternalIP);
   }
 
+  private static PeeringState peeringState(@Nullable String systemNetwork, String systemProjectId,
+                                           Network networkInfo, Compute compute) throws IOException {
+    // systemNetwork can be null when CDAP is not running on GCP for example CDAP running in sandbox environment
+    if (systemNetwork == null) {
+      return PeeringState.NONE;
+    }
+
+    Network systemNetworkInfo = getNetworkInfo(systemProjectId, systemNetwork, compute);
```

Comment:
This requires permission on the network which is acceptable but will not be present on existing setups and will require backfill.

Reply:
@sagarkapare : Thinking more about this permission issue change it has a much wider impact. This requires the management SA to have network permission on the tp network. In Beta it does not which means that for existing customer if we roll out this change the dataproc provisioning will start failing for all instances till the permissions are backfilled.